### PR TITLE
Use new model for flaw acknowledgments when saving data via bbsync

### PR DESCRIPTION
**Pull request by jsvob**
Follow up to https://github.com/RedHatProductSecurity/osidb/pull/273 and https://github.com/RedHatProductSecurity/osidb/pull/265

Related to OSIDB-1004 ("Implement BBSync part for FlawAcknowledgment")

### CODE REVIEW

**Comment by jsvob** [comment]
Blocked on https://github.com/RedHatProductSecurity/osidb/pull/265

File: apps/bbsync/tests/test_query.py
Comment (jobselko) [inline apps/bbsync/tests/test_query.py:254]: `name` and `affiliation` are probably wrong. `name` can be e.g. "Carl von Bahnhof" and `affiliation` can be e.g. "Google Security Team". (I think "von Bahnhof" is a part of the name, not an organization. This problem occurs more times.)

File: apps/bbsync/tests/test_query.py
Comment (jobselko) [inline apps/bbsync/tests/test_query.py:1]: Could you please update `test_schema`? It tests complex flaw SRT notes generation, and it still contains acknowledgment from `FlawMetaFactory`.

**Review by jobselko — CHANGES_REQUESTED** [review]

File: apps/bbsync/tests/test_query.py
Comment (jsvob) [reply apps/bbsync/tests/test_query.py]: Added names that should be more descriptive for demo purposes.

File: apps/bbsync/tests/test_query.py
Comment (jsvob) [reply apps/bbsync/tests/test_query.py:1]: Updated the test function to use the new model and also added a test to make sure the test didn't run on empty data (before the fix it ran on empty data and didn't fail).

Comment (jobselko) [inline apps/bbsync/tests/test_query.py:1]: You forgot to update `acknowledgments` in `srtnotes` (used in `FlawFactory`'s `meta_attr)` in `test_schema`.

Comment (jsvob) [reply apps/bbsync/tests/test_query.py:1]: Fixed. Thanks.

Comment (jobselko) [reply apps/bbsync/tests/test_query.py:1]: I am unsure if I understand the second part of your sentence. I would like to discuss it with you.

Comment (jsvob) [reply apps/bbsync/tests/test_query.py:1]: I didn't notice the original mistake you noticed here. A part of the reason was that I didn't pay enough attention (not fixable by tests), and a part of the reason was that the test ran and didn't test FlawAcknowledgment srtnotes generation at all. So the newly added code at the end of `test_schema` tests that the main part of the test before it doesn't run on no data. Will explain more on 1:1.

Comment (jsvob) [reply apps/bbsync/tests/test_query.py:1]: Aha, the sentence `# Additionally, ensure that the validations were not run on empty data.` was missing the word `not`. Sorry for the confusion.

**Review by Elkasitu — APPROVED** [review]
LGTM but I would squash these commits

**Review by jobselko — APPROVED** [review]

**Pull request merged**
